include/aws/io/socket.h

```diff
@@ -176,11 +176,17 @@ AWS_IO_API int aws_socket_connect(
  */
 AWS_IO_API int aws_socket_bind(struct aws_socket *socket, const struct aws_socket_endpoint *local_endpoint);
 
+/**
+ * TCP/UDP only. Get bound port number.
+ */
+AWS_IO_API int aws_socket_get_bound_address(struct aws_socket *socket, struct aws_socket_endpoint *local_address);
```

Comment:
I think the socket can be const.

Comment seems inaccurate given that it's getting more than the port.

Reply:
I thinking about const but unfortunalety windows implementation has a `socket->state = ERRORED;`
I copied that behaviour from aws_socket_bind() implementation.

Reply:
Note updated in dfab913a3c6fead5d66c9f498fcbac74d24e4c34